libuavcan/include/uavcan/bus/can.hpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ */
+/** @file */
+
+#ifndef UAVCAN_BUS_CAN_HPP_INCLUDED
+#define UAVCAN_BUS_CAN_HPP_INCLUDED
+
+#include <array>
+
+#include "uavcan/uavcan.hpp"
+
+namespace uavcan
+{
+/**
+ * @namespace bus
+ * Contains bus-specific namespaces.
+ */
+namespace bus
+{
+/**
+ * @namespace CAN
+ * Types for working with UAVCAN on a Controller Area Network.
+ */
+namespace CAN
+{
+/**
+ * @namespace TypeFd
+ * Properties of an ISO compliant CAN-FD bus.
+ */
+namespace TypeFd
+{
+/**
+ * The maximum size of a data frame for this bus.
+ */
+constexpr static size_t max_frame_size_bytes = 64;
+
+/** The size of the tail byte, in bytes. */
+constexpr static size_t tail_byte_size_bytes = 1;
+
+/** The number of bytes in the transfer CRC. */
+constexpr static size_t transfer_crc_size_bytes = 2;
```

Comment:
What is the point of specifying the CRC size and tail bytes on the per-transport basis? These parameters are identical for both CAN transports, and other transports are far more different than can be described using this mechanism.

Reply:
okay. will promote to namespace.